src/Compilers/VisualBasic/Test/Semantic/Semantics/RequiredMembersTests.vb

```diff
@@ -0,0 +1,1842 @@
+
+            comp.AssertTheseDiagnostics(<expected><![CDATA[
+BC37325: 'System.Runtime.CompilerServices.RequiredMemberAttribute' is reserved for compiler usage only.
+<RequiredMember> ' 1
+ ~~~~~~~~~~~~~~
+BC37325: 'System.Runtime.CompilerServices.RequiredMemberAttribute' is reserved for compiler usage only.
+    <RequiredMember> ' 2
+     ~~~~~~~~~~~~~~
+BC37325: 'System.Runtime.CompilerServices.RequiredMemberAttribute' is reserved for compiler usage only.
+    <RequiredMember> ' 3
+     ~~~~~~~~~~~~~~]]></expected>)
+        End Sub
+
+        <Fact>
+        Public Sub TupleWithRequiredFields()
+            Dim csharpComp = CreateCSharpCompilation("
+namespace System
+{
+    public struct ValueTuple<T1, T2>
+    {
+        public required T1 Item1;
+        public required T2 Item2;
+        public required int AnotherField;
+        public required int Property { get; set; }
+
+        public ValueTuple(T1 item1, T2 item2)
+        {
+            this.Item1 = item1;
+            this.Item2 = item2;
+        }
+
+        public static bool operator ==(ValueTuple<T1, T2> t1, ValueTuple<T1, T2> t2)
+            => throw null;
+        public static bool operator !=(ValueTuple<T1, T2> t1, ValueTuple<T1, T2> t2)
+            => throw null;
+
+        public override bool Equals(object o)
+            => throw null;
+        public override int GetHashCode()
+            => throw null;
+    }
+
+    namespace Runtime.CompilerServices
+    {
+        [AttributeUsage(AttributeTargets.Class | AttributeTargets.Struct | AttributeTargets.Field | AttributeTargets.Property, Inherited = false, AllowMultiple = false)]
+        public sealed class RequiredMemberAttribute : Attribute
+        {
+            public RequiredMemberAttribute()
+            {
+            }
+        }
+
+        [AttributeUsage(AttributeTargets.All, AllowMultiple = true, Inherited = false)]
+        public sealed class CompilerFeatureRequiredAttribute : Attribute
+        {
+            public CompilerFeatureRequiredAttribute(string featureName)
+            {
+                FeatureName = featureName;
+            }
+            public string FeatureName { get; }
+            public bool IsOptional { get; set; }
+        }
+    }
+    namespace Diagnostics.CodeAnalysis
+    {
+        [AttributeUsage(AttributeTargets.Constructor, Inherited = false, AllowMultiple = false)]
+        public sealed class SetsRequiredMembersAttribute : Attribute
+        {
+            public SetsRequiredMembersAttribute()
+            {
+            }
+        }
+    }
+}
+", referencedAssemblies:=Basic.Reference.Assemblies.Net461.All)
+
+            ' Using Net461 to get a framework without ValueTuple
+
+            Dim comp = CreateCompilation("
+Class C
+    Sub Main()
+        Dim t1 = (1, 2)
+        Dim t2 As (Integer, Integer) = (3, Nothing)
+        Dim t3 = New (Integer, Integer)(4, 5)
+        Dim t4 = new System.ValueTuple(Of Integer, Integer)(6, 7)
```

Comment:
>Dim t4 = new System.ValueTuple(Of Integer, Integer)(6, 7) [](http://example.com/codeflow?start=8&length=57)

Consider also testing ```new System.ValueTuple(of integer, integer)()```

Reply:
Plus ```Dim t2 As (Integer, Integer) = Nothing``` and ```Dim t2 As System.ValueTuple(of integer, integer) = Nothing```

Reply:
BTW, it looks like the following is a valid C# code. Is this expected?
```
S1 x = default;

struct S1
{
    public required int F;
}
```

Reply:
> BTW, it looks like the following is a valid C# code. Is this expected?

Yes. I forgot to explicitly document that in the spec, but it is expected. I'll do that.